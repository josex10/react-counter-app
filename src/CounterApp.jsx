import { useState } from 'react';
import PropTypes from 'prop-types';
export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = () => setCounter(counter + 1);
  const handleSubtract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);
  return (
    <>
      <h1 aria-label="heading-title">Counter App</h1>
      <h2 aria-label="heading-counter">{counter}</h2>

      <button aria-label="btn-sum" onClick={handleAdd}>
        +1
      </button>
      <button aria-label="btn-sub" onClick={handleSubtract}>
        -1
      </button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number
};

CounterApp.defaultProps = {
  value: 0
};
