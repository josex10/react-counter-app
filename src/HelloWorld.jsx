import PropTypes from 'prop-types';

export const HelloWorld = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

HelloWorld.propTypes = {
  subTitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

HelloWorld.defaultProps = {
  name: 'Jose Manuel Badilla Porras - default',
  subTitle: 'Without subtitle prop - default'
};
