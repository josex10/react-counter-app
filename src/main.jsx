import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorld } from './HelloWorld';
import { CounterApp } from './CounterApp';
import './style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <HelloWorld />
  </React.StrictMode>
);
