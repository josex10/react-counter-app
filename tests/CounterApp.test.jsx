import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Testing the component CounterApp', () => {
  const initialValue = 100;

  test('Should be a match with the snapshot', () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test('Should be show the initial value on 100', () => {
    render(<CounterApp value={initialValue} />);
    expect(screen.getByText(initialValue)).toBeTruthy();
  });

  test('Testing the button sum', () => {
    //Arrange
    render(<CounterApp value={initialValue} />);

    //Act
    fireEvent.click(screen.getByRole('button', { name: 'btn-sum' }));
    const expectedValue = initialValue + 1;

    //Assert
    expect(
      screen.getByRole('heading', { name: 'heading-counter' }).innerHTML
    ).toBe(String(expectedValue));
  });

  test('Testing the button sub', () => {
    //Arrange
    render(<CounterApp value={initialValue} />);

    //Act
    fireEvent.click(screen.getByRole('button', { name: 'btn-sub' }));
    const expectedValue = initialValue - 1;

    //Assert
    expect(
      screen.getByRole('heading', { name: 'heading-counter' }).innerHTML
    ).toBe(String(expectedValue));
  });

  test('Testing the button reset', () => {
    //Arrange
    render(<CounterApp value={initialValue} />);

    //Act
    fireEvent.click(screen.getByRole('button', { name: 'btn-sub' }));
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

    //Assert
    expect(
      screen.getByRole('heading', { name: 'heading-counter' }).innerHTML
    ).toBe(String(initialValue));
  });
});
