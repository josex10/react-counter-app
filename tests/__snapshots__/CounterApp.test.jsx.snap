// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the component CounterApp Should be a match with the snapshot 1`] = `
<div>
  <h1
    aria-label="heading-title"
  >
    Counter App
  </h1>
  <h2
    aria-label="heading-counter"
  >
    100
  </h2>
  <button
    aria-label="btn-sum"
  >
    +1
  </button>
  <button
    aria-label="btn-sub"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
